checks.d/jenkins.py

```diff
@@ -27,17 +28,46 @@ def __init__(self, name, init_config, agentConfig):
         AgentCheck.__init__(self, name, init_config, agentConfig)
         self.high_watermarks = {}
 
+    def _parse_timestamp(self, dir_name):
+        if os.path.exists(os.path.join(dir_name, 'jenkins_build.tar.gz')):
+            raise Skip('the build has already been archived', dir_name)
+
+        # Read the build.xml metadata file that Jenkins generates
+        build_file = os.path.join(dir_name, 'build.xml')
+
+        if not os.access(build_file, os.R_OK):
+            self.log.debug("Can't read build file at %s" % (build_file))
+            raise Exception("Can't access build.xml at %s" % (build_file))
+        else:
+            tree = ElementTree()
+            tree.parse(build_file)
+            timestamp = tree.find('timestamp')
+            if timestamp is None or not timestamp.text:
+                raise Skip('the timestamp cannot be found', dir_name)
+            else:
+                return timestamp.text
+
     def _extract_timestamp(self, dir_name):
         if not os.path.isdir(dir_name):
             raise Skip('its not a build directory', dir_name)
 
+        dir_basename = os.path.basename(dir_name)
         try:
-            # Parse the timestamp from the directory name
-            date_str = os.path.basename(dir_name)
-            time_tuple = time.strptime(date_str, self.datetime_format)
-            return time.mktime(time_tuple)
+            dir_basename = int(dir_basename)
         except ValueError:
-            raise Exception("Error with build directory name, not a parsable date: %s" % (dir_name))
+            pass
+        if isinstance(dir_basename, int):
+            # Parse the timestamp from the build.xml
```

Comment:
Do we need to parse the timestamp in this case ?

If i understand, we need this timestamp just to know what was the latest build the agent analyzed. But we could just use the build number for this as we can assume that it's an always increasing number.